doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
+
+Splitting patterns can have more than 2 names, for example :g:`[H1 H2 H3]`
+matches :g:`A /\ B /\ C`.  Patterns can be nested:
+:g:`[[Ha|Hb] H]` can be used to split :g:`(A \/ B) /\ C`.
+
+Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
+use the :tacn:`split` tactic to replace the current goal with subgoals :g:`A` and :g:`B`.
+For a goal :g:`A \/ B`, use :g:`apply or_introl` to replace the current goal with :g:`A`, or
+:g:`apply or_intror` to replace the current goal with :g:`B`.
+
+The other splitting patterns described below are not frequently used.
+
+These patterns can be used when the hypothesis is an equality:
+
+* :n:`->` - replaces the right-hand side of the hypothesis with the left-hand
+  side of the hypothesis in the conclusion of the goal; the hypothesis is
+  cleared; if the left-hand side of the hypothesis is a variable, it is
+  substituted everywhere in the context and the variable is removed.
+* :n:`<-` - similar to :n:`->`, but replaces the left-hand side of the hypothesis
+  with the right-hand side of the hypothesis.
+* :n:`[= @intropatterns ]` - If the product is over an equality type,
+  applies either :tacn:`injection` or :tacn:`discriminate`
+  instead of :tacn:`destruct`; if :tacn:`injection` is applicable, the patterns
+  :n:`{+, p}` are used on the hypotheses generated by :tacn:`injection`; if the
+  number of patterns is smaller than the number of hypotheses generated, the
+  pattern :n:`?` is used to complete the list.
+
+**need example for [= ... ]**
+
+* :n:`( @simple_intropattern , ... , @simple_intropattern )` - matches
+  a product over an inductive type :g:`I` with a single
+  constructor that itself has at least `n` arguments.  It performs a case
+  analysis over the hypothesis, as :n:`destruct` would, and applies the
+  patterns to the arguments of the constructor of :g:`I`.  Note that
+  that :n:`({+ p})` is equivalent to :n:`[{+ p}]`);
+* :n:`( @simple_intropattern & ... & @simple_intropattern )` - :n:`({+& p})`
+  is a shortcut for matching with
+  :n:`(p,( ... ,( ..., p ) ... ))`.  It expects the hypothesis to be a sequence of
+  right-associative binary inductive constructors such as :g:`conj` or
+  :g:`ex_intro`; for instance, a hypothesis with type
+  :g:`A /\(exists x, B /\ C /\ D)` can be introduced via pattern
+  :n:`(a & x & b & c & d)`;
+
+
+**need example for & ... &**
+
+**A /\\ (exists x, B /\\ C /\\ D) is not valid for (a & x & b & c & d)**
+
+* :n:`*` - introduces one or more quantified variables from the result
+* :n:`**` - introduces one or more quantified variables or hypotheses from the result until there are
+  no more quantified variables or implications (:g:`->`).  :g:`intros **` is equivalent
+  to :g:`intros`.
+
+* :n:`%term` - a pattern :n:`p{+ %term}` first applies :n:`{+ term}`
+  on the hypothesis to be introduced (as in :n:`apply {+, term}`) prior to the
+  application of the introduction pattern :n:`p`;
```

Comment:
Note in passing. This would be nice to use actual tokens more systematically instead of `p`